packages/connectors/ledger-connector/src/hid/connector.ts

```diff
@@ -0,0 +1,80 @@
+import invariant from 'tiny-invariant';
+import { AbstractConnector } from '@web3-react/abstract-connector';
+import type { ConnectorUpdate } from '@web3-react/types';
+import { checkError, isHIDSupported } from './helpers';
+
+import type { LedgerHQProvider } from './provider';
+
+type LedgerHQConnectorArguments = {
+  chainId: number;
+  url: string;
+};
+
+export class LedgerHQConnector extends AbstractConnector {
+  public provider?: LedgerHQProvider;
+
+  public url: string;
+
+  public chainId: number;
+
+  constructor({ chainId, url }: LedgerHQConnectorArguments) {
+    super({ supportedChainIds: [chainId] });
+
+    this.chainId = chainId;
+    this.url = url;
+
+    this.handleDisconnect = this.handleDisconnect.bind(this);
+  }
+
+  private handleDisconnect(): void {
+    this.emitDeactivate();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  public isSupported(): boolean {
+    return isHIDSupported();
+  }
+
+  public async getProviderInstance(): Promise<LedgerHQProvider> {
+    // eslint-disable-next-line no-shadow
+    const { LedgerHQProvider } = await import('./provider');
+    const Provider = new LedgerHQProvider(this.url, this.chainId);
+
+    return Provider;
```

Comment:
1. it should not be capitalised as it is neither component or class 
2. it can be simpler
```suggestion
    return new LedgerHQProvider(this.url, this.chainId);
```

Reply:
3. should not it be cached?

Reply:
1,2 – fixed, but there are a lot of other eslint complains to fix, since the code was just copy-pasted from other repos.
3 – it is cached like this: https://github.com/lidofinance/reef-knot/blob/d92eac2ba6308625e49f64200a3d5008f3ae1b5e/packages/connectors/ledger-connector/src/hid/connector.ts#L46-L48